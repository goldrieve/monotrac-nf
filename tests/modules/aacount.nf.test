nextflow_process {

    name "Test Process AA_COUNT"
    script "main.nf"
    process "AA_COUNT"

    test("Should run AA count") {

        when {
            process {
                """
                input[0] = ["SVP_test", file("${projectDir}/tests/data/SVP_test_AA.fas")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}