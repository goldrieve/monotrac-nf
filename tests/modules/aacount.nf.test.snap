{
    "Should run AA count": {
        "content": [
            {
                "0": [
                    "SVP_test_AA_counts.csv:md5,f9fc7fe7fda1e29eefacef16a320e9d4"
                ],
                "1": [
                    [
                        "SVP_test",
                        "SVP_test_AA_counts.csv:md5,f9fc7fe7fda1e29eefacef16a320e9d4"
                    ]
                ],
                "counts": [
                    "SVP_test_AA_counts.csv:md5,f9fc7fe7fda1e29eefacef16a320e9d4"
                ],
                "sample_counts": [
                    [
                        "SVP_test",
                        "SVP_test_AA_counts.csv:md5,f9fc7fe7fda1e29eefacef16a320e9d4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T12:36:59.694763695"
    }
}